### PR TITLE
feature: add ability to export a json schema for opslevel.yaml

### PR DESCRIPTION
**Pull request by rocktavious**
Add the command `opslevel export schema` which emits a json schema of the structure which mirrors opslevel.yml

### CODE REVIEW

File: src/cmd/opslevel.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	Path     string `json:"path,omitempty"`
+	Provider string `json:"provider"` // TODO: this should be an enum
+}
+
+type OpsLevelServiceDependancy struct {
+	Alias string `json:"alias"`
+}
+
+type OpsLevelService struct {
+	Name         string                      `json:"name"`
```
Comment (maxim-opslevel) [inline src/cmd/opslevel.go:35]: why no `omitempty` on this field?

File: src/cmd/opslevel.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	Name        string                `json:"name"`
+	Category    opslevel.ToolCategory `json:"category"`
+	Url         string                `json:"url"`
+	Environment string                `json:"environment,omitempty"`
+}
+
+type OpsLevelRepository struct {
+	Name     string `json:"name"`
+	Path     string `json:"path,omitempty"`
+	Provider string `json:"provider"` // TODO: this should be an enum
```
Comment (maxim-opslevel) [inline src/cmd/opslevel.go:27]: You're missing

DisplayName

**Comment by maxim-opslevel** [comment]
This works for a `Service` opslevel.yml but not a `Repository` [opslevel.yml](https://www.opslevel.com/docs/api/opslevel-yml/#repository). I think we need to allow for the creation of a schema for this as well.

File: src/cmd/opslevel.go (hunk 1 above)
Comment (rocktavious) [reply src/cmd/opslevel.go:35]: Name is required

File: src/cmd/opslevel.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	Aliases      []string                    `json:"aliases,omitempty"`
+	Tags         []OpsLevelTag               `json:"tags,omitempty"`
+	Tools        []OpsLevelTool              `json:"tools,omitempty"`
+	Repositories []OpsLevelRepository        `json:"repositories,omitempty"`
+	Dependencies []OpsLevelServiceDependancy `json:"dependencies,omitempty"`
+}
+
+type OpsLevelConfig struct {
+	Version int             `json:"version"`
+	Service OpsLevelService `json:"service"`
```
Comment (rocktavious) [inline src/cmd/opslevel.go:52]: @maxim-opslevel i think we just add this here and it builds 1 schema that works for both
```
Repository OpsLevelRepository `json:"repository"`
```

But i guess that question becomes - "can you define both a repository and a service inside 1 Git repo inside the same opslevel.yml?"

File: src/cmd/opslevel.go (hunk 2 above)
Comment (rocktavious) [reply src/cmd/opslevel.go]: Added

File: src/cmd/opslevel.go (hunk 3 above)
Comment (rocktavious) [reply src/cmd/opslevel.go]: Added "one_of" requirement

**Comment by rocktavious** [comment]
json schema is now available via https://app.opslevel.com/public/schemas/opslevel.schema.yml

**Pull request closed without merging**